pdaltools/add_points_in_pointcloud.py

```diff
@@ -280,16 +289,28 @@ def add_points_from_geometry_to_las(
     if len(unique_geom_type) != 1:
         raise ValueError("Several geometry types found in geometry file. This case is not handled.")
 
-    if unique_geom_type == ["Point"] or unique_geom_type == ["MultiPoint"]:
-        # Add the Z dimension from the 'RecupZ' property
-        gdf["geometry"] = gdf.apply(
-            lambda row: Point(row["geometry"].x, row["geometry"].y, row[altitude_column]), axis=1
-        )
-        # If the geometry type is Point, use the points directly
-        points_gdf = gdf[["geometry"]]
-    elif unique_geom_type == ["LineString"] or unique_geom_type == ["MultiLineString"]:
+    if unique_geom_type in ["Point", "MultiPoint"]:
+        if altitude_column:
+            # Add the Z dimension from the 'RecupZ' property
+            gdf["geometry"] = gdf.apply(
+                lambda row: Point(row["geometry"].x, row["geometry"].y, row[altitude_column]), axis=1
+            )
+            # If the geometry type is Point, use the points directly
+            points_gdf = gdf[["geometry"]]
+        else:
+            # Add the Z dimension from the Geometry Z
+            gdf["geometry"] = gdf.apply(
+                lambda row: Point(row["geometry"].x, row["geometry"].y, row["geometry"].z), axis=1
+            )
+            # If the geometry type is Point, use the points directly
+            points_gdf = gdf[["geometry"]]
+    elif unique_geom_type in ["LineString", "MultiLineString"]:
+        gdf = clip_3d_lines_to_tile(gdf, input_las, spatial_ref, tile_width)
         # If the geometry type is LineString, generate 3D points
-        points_gdf = generate_3d_points_from_lines(gdf, spacing, altitude_column)
+        if spacing <= 0:
+            raise ValueError("The parameters spacing <= 0.")
```

Comment:
tu peux être plus explicite ici : 
par exemple 
```suggestion
            raise NotImplementedError(f"add_points_from_geometry_to_las requires spacing > 0 to run on (Multi)LineString geometries, but the values provided are geometry type: {unique_geom_type} and spacing = {spacing} ")
```